### PR TITLE
feat: add drift manifest generation

### PR DESCRIPTION
**Pull request by viacheslav-rostovtsev**
drift manifest maps the correspondence between proto description of the package and the generated code. 
schema: https://github.com/googleapis/googleapis/blob/master/gapic/metadata/gapic_metadata.proto

### CODE REVIEW

File: gapic-generator/lib/gapic/generators/default_generator.rb [hunk 1]
```diff
@@ -77,6 +79,10 @@ def generate gem_presenter: nil
         files << g("gem/yardopts.erb",    ".yardopts",                    gem: gem)
         files << g("gem/license.erb",     "LICENSE.md",                   gem: gem)
         files << g("gem/entrypoint.erb",  "lib/#{gem.name}.rb",           gem: gem)
+        files << generate_nontemplate_file(
+          filename: "gapic_metadata.json",
+          content: JSON::pretty_generate(gem.first_package_drift_manifest)
```
Comment (dazuma) [inline gapic-generator/lib/gapic/generators/default_generator.rb:84]: Maybe consider just creating a template with this logic, so we don't need a separate generation method.

Comment (dazuma) [reply gapic-generator/lib/gapic/generators/default_generator.rb:84]: In fact, what we'd do elsewhere is actually create another presenter method, say, `first_package_drift_manifest_json`, which does the JSON pretty_generate, and then just render that directly in a template.

Comment (viacheslav-rostovtsev) [reply gapic-generator/lib/gapic/generators/default_generator.rb:84]: What benefits does the template layer bring to justify routing non-template files through it?

Comment (viacheslav-rostovtsev) [reply gapic-generator/lib/gapic/generators/default_generator.rb]: fixed

**Review by dazuma — APPROVED** [review]

**Pull request merged**